mongo/integration/sessions_prose_test.go

```diff
@@ -0,0 +1,22 @@
+package integration
+
+import (
+	"errors"
+	"testing"
+
+	"go.mongodb.org/mongo-driver/internal/testutil/assert"
+	"go.mongodb.org/mongo-driver/mongo/integration/mtest"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+func TestSessionsProse(t *testing.T) {
```

Comment:
@benjirewis Is this test redundant now with the new test cases in `TestClientSession`?

Reply:
It is redundant with the "both true" subtest in `TestClientSession/causal_consistency_and_snapshot`. 

I removed that subtest in favor of this prose test and left a comment pointing to `TestSessionsProse`. That way, it will be more obvious that we've actually implemented this prose test since it's in `sessions_prose_test`.